### ISSUE TITLE
Payment Gateway Pay by Check

### ISSUE THREAD

**Issue by joshuarlowry** [firesale/firesale#135]
I need to have the ability to allow users to pay by check so I created a new gateway.  What are the allowable merchant responses?  Can they be anything?  I need it to indicate that the user decided to pay with a check and therefore will need to be manually verified.

``` php
class Merchant_dummy extends Merchant_driver
{
    public $required_fields = array('amount', 'currency_code', 'transaction_id', 'reference');

    public $settings = array();

    public function _process($params)
    {
            return new Merchant_response('authorized', 'The transaction was authorized', null, $params['amount']);
    }
}
```

**Comment by jHoldroyd** [comment]
The available status options are:
- failed
- declined
- mismatch
- authorized

These all have corresponding functions in front_cart that will be called accordingly.

**Comment by joshuarlowry** [comment]
I think I'm unsure then of what this should be returning.  What I'm trying to set is that the order is something like payment pending.

**Comment by chrisnharvey** [comment]
Payment pending is something I'm actually working on right now. Will keep you updated with the progress.

**Comment by chrisnharvey** [comment]
I'll get this fixed as part of #233
